### PR TITLE
Split large compilation units

### PR DESCRIPTION
**Pull request by no92**

### CODE REVIEW

**Comment by avdgrinten** [comment]
:+1: This is a good idea (didn't review everything carefully but overall it looks good).

**Review by Dennisbonke — APPROVED** [review]
LGTM

File: posix/subsystem/src/supercalls.cpp [hunk 1]
```diff
@@ -0,0 +1,610 @@
+#include "gdbserver.hpp"
+#include "supercalls.hpp"
```
Comment (qookei) [inline posix/subsystem/src/supercalls.cpp:2]: This file also handles other observations apart from supercalls (faults and such), perhaps we could pick a better name.

**Review by qookei — COMMENTED** [review]
Looks good for the most part

**Comment by no92** [comment]
Addressed the comments raised above.

**Review by qookei — APPROVED** [review]

**Pull request merged**